vernac/vernacentries.ml

```diff
@@ -2679,22 +2679,29 @@ and vernac_load ~verbosely fname =
     CErrors.user_err Pp.(str "Files processed by Load cannot leave open proofs.");
   ()
 
-and interp_control ?proof ~st v = match v with
-  | { v=VernacExpr (atts, cmd) } ->
-    let before_univs = Global.universes () in
-    let pstack = interp_expr ?proof ~atts ~st cmd in
-    if before_univs == Global.universes () then pstack
-    else Option.map (Vernacstate.LemmaStack.map_top_pstate ~f:Proof_global.update_global_env) pstack
-  | { v=VernacFail v } ->
-    with_fail ~st (fun () -> interp_control ?proof ~st v);
+and interp_control_flag ~vernac (f : control_flag) ~st
```

Comment:
This isn't mutually recursive with the rest afaict

Reply:
It is not, indeed I forgot to move it upwards [I kept it close for the refactoring as to see both the old and the new].

Moved.